### DISCUSSION TITLE
Is there GitHub rest API that we could use to accept invitations to org/team?

### DISCUSSION THREAD

**Discussion by shd101wyy** [community/community#29606] [Apps, API and Webhooks] [answered]
Hi there,

Is there GitHub rest API that we could use to accept invitations to org/team?

Right now I only found API to accept the Repository Invitations, but not the org/team invitations.

Thank you

**Comment by luckylittle** [comment, marked as answer]
@shd101wyy I believe this is simply done by opening the URL `https://github.com/orgs/<ORG_NAME>/invitation?via_email=1` while logged in as the user who was invited. You can also accept the invitation by visiting the organization page directly at `https://github.com/<ORG_NAME>/`.

_Note: Please accept the answer if it answers your question, thanks_

**Reply by shd101wyy** [reply]
Thanks @luckylittle . The problem is I don’t have permission to login (it’s a company account). I only have the personal access token of that account to call web APIs.

**Reply by shd101wyy** [reply]
And we want to implement a workflow to automatically accept invitations to repository/team/org for that company account

**Reply by luckylittle** [reply]
> Thanks @luckylittle . The problem is I don’t have permission to login (it’s a company account). I only have the personal access token of that account to call web APIs.

I believe something like this might work:

1. GET method to get `INVITATION_ID`:

```bash
curl \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <PERSONAL_ACCESS_TOKEN_OF_THE_INVITED_ACCOUNT>" \
  https://api.github.com/orgs/<ORGANIZATION>/invitations
```

2. POST method:

```bash
curl \
  -X POST \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token <PERSONAL_ACCESS_TOKEN_OF_THE_INVITED_ACCOUNT>" \
  https://api.github.com/orgs/<ORGANIZATION>/invitation/<INVITATION_ID_FROM_STEP_1> \
  -d '{"email":"octocat@github.com"}'
```

**Reply by luckylittle** [reply]
Send me an invite to some arbitrary/temporary organisation and I can capture the HTTP flow to see how it's done

**Reply by shd101wyy** [reply]
Thanks @luckylittle , I will give it a try.

invited you to github.com/0xGG

**Reply by luckylittle** [reply]
@shd101wyy 
I think it is simpler than that. You need to log in first with your token and then just go to that URL with the same cookie:

```bash
#!/bin/zsh

curl -v \
  -c cookie.txt \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: token <REMOVED>" \
  https://api.github.com/user

curl -v \
  -b cookie.txt \
  -X POST \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: token <REMOVED>" \
  https://github.com/orgs/0xGG/invitation?via_email=1
```

**Comment by shd101wyy** [comment]
Thanks @luckylittle!

your approach works if you already know which org invited you in advance. 

But for my case, I wouldn’t know which organization invited me. So I would assume I will need to get a list of invitations to orgs, then accept them one by one 🤔

Thanks!

**Reply by shd101wyy** [reply]
Found a solution.

I can [List organization memberships for the authenticated user
](https://docs.github.com/en/rest/orgs/members#list-organization-memberships-for-the-authenticated-user) then [Update an organization membership for the authenticated user](https://docs.github.com/en/rest/orgs/members#update-an-organization-membership-for-the-authenticated-user) :+1:
